### DISCUSSION TITLE
Plotting cos theta distribution histogram instead of theta distribution

### DISCUSSION THREAD

**Discussion by abhika2s** [MadAnalysis/madanalysis5#189] [Q&A] [answered]
In MadAnalysis plot function, there is an option to plot THETA but no option to plot cos THETA. How can we customize the code to give cos THETA distribution instead of THETA?

**Comment by BFuks** [comment]
Hi,

You can modify the code generated by the theta distribution. That’s the easiest. The other option would be to code everything from scratch in the expert mode of MadAnalysis5.

Cheers,

Benjamin

**Reply by abhika2s** [reply]
Hi, Benjamin. Can you give a little bit of details about the PATH of the generated code for theta distribution? In the output folder, I can see some codes for plotting the histos. But can we modify them from theta to cos theta? The evenly spaced theta intervals won't be evenly spaced for cos theta.

**Comment by BFuks** [comment]
Hi,

The code is located in ```Build/SampleAnalyzer/User/Analyzer/```. Once modified, then you can go back to ```Build```, recompile and rerun.

Cheers,

Benjamin

**Reply by abhika2s** [reply]
I did the modification in the code, however when I tried the command make in the Build directory, it returned with following error:

-e --------------------------------------------------------
-e                 Building MadAnalysis Job                
-e --------------------------------------------------------
-e The shared library /tools/SampleAnalyzer/Lib/libcommons_for_ma5.so is not found
-e  1) Please check that MadAnalysis 5 is installed in the folder : 
-e  2) Launch MadAnalysis 5 in normal mode in order to build this library.
-e 
make: *** [Makefile:53: library_check] Error 1

Is there any other way to recompile this particular analysis?

**Comment by BFuks** [comment]
Hi Abhik,

Have you run the setup script (setup.sh, located in Build)?

Cheers,

Benjamin

**Reply by abhika2s** [reply]
I tried to run the setup script file using:

sh setup.sh

as well as

chmod + x setup.sh
./setup.sh

Both returned:

setup.sh: 30: [[: not found

I then tried:

bash setup.sh

It returned:

--------------------------------------------------------
    Your environment is properly configured for MA5     
--------------------------------------------------------

But both didn't compile the script. And same goes for setup.csh

**Comment by BFuks** [comment, marked as answer]
Have you tried “source setup.sh” ?

Cheers,

Benjamin
